views/repository.py

```diff
@@ -98,12 +98,22 @@ def all(resource, key, value):
                 u.first_name,
                 u.last_name,
                 u.username,
-                c.label
+                c.id cat_id,
+                c.label cat_label,
+                pt.tag_id tag_id,
+                pt.post_id post_id, 
+                t.id tag_id, 
+                t.label tag_label
             FROM posts p
             JOIN users u
                 ON u.id = p.user_id
             JOIN categories c
                 on c.id = p.category_id
+            JOIN posttags pt
```

Comment:
Try this with LEFT JOIN instead, posts with no corresponding record on posttags table are not being returned.